projects/plugins/jetpack/modules/subscriptions/subscribe-modal/subscribe-modal.php

```diff
@@ -0,0 +1,196 @@
+				'post_content' => get_subscribe_template_content(),
+				'post_status'  => 'publish',
+				'post_author'  => 1,
+				'post_type'    => 'wp_template_part',
+				'scope'        => array(),
+				'tax_input'    => array(
+					'wp_theme' => get_option( 'stylesheet' ),
+				),
+			);
+			wp_insert_post( $template );
+		}
+	}
+}
+add_action( 'wp_loaded', 'wpcom_create_subscribe_template' );
+
+/**
+ * Adds modal with Subscribe Modal content.
+ *
+ * @return void
+ */
+function wpcom_add_subscribe_modal_to_frontend() {
+	if ( should_enable_subscriber_modal() && ! is_admin() ) {
+		$posts = get_posts(
+			array(
+				'post_type'   => 'wp_template_part',
+				'post_status' => 'publish',
+				'numberposts' => -1,
+			)
+		);
+
+		$subscribe_template = reset(
+			array_filter(
+				$posts,
+				function ( $post ) {
+					return $post->post_name === get_subscribe_template_slug();
+				}
+			)
+		);
+
+		$blocks = parse_blocks( $subscribe_template->post_content );
+
+		/*
+		 * Jetpack requires escaping html for the render_block() call below.
+		 * I'm using wp_kses() and specifying the allowed tags.
+		 * But this is fragile: users can edit the Subscribe Modal
+		 * template and add tags that are not allowed here.
+		 */
+		$allowed_html = array(
+			'div'    => array(
+				'class' => array(),
+				'style' => array(),
+			),
+			'h2'     => array(
+				'class' => array(),
+				'style' => array(),
+			),
+			'p'      => array(
+				'type'  => array(),
+				'class' => array(),
+				'style' => array(),
+				'id'    => array(),
+			),
+			'form'   => array(
+				'action'                 => array(),
+				'method'                 => array(),
+				'accept-charset'         => array(),
+				'data-blog'              => array(),
+				'data-post_access_level' => array(),
+				'id'                     => array(),
+			),
+			'label'  => array(
+				'id'    => array(),
+				'for'   => array(),
+				'class' => array(),
+			),
+			'input'  => array(
+				'id'          => array(),
+				'type'        => array(),
+				'name'        => array(),
+				'value'       => array(),
+				'class'       => array(),
+				'style'       => array(),
+				'placeholder' => array(),
+				'required'    => array(),
+			),
+			'button' => array(
+				'type'  => array(),
+				'class' => array(),
+				'style' => array(),
+			),
+		);
+
+		?>
+			<div id="wpcom-subscribe-modal">
+				<div class="modal-content">
+					<?php
+					foreach ( $blocks as $block ) {
+						echo wp_kses( render_block( $block ), $allowed_html );
+					}
+					?>
+					<div id="close"><a href="#">Close</a></div>
+				</div>
+			</div>
+		<?php
+	}
+}
+add_action( 'wp_footer', 'wpcom_add_subscribe_modal_to_frontend' );
+
+/**
+ * Enqueues JS to load modal.
+ *
+ * @return void
+ */
+function wpcom_enqueue_subscribe_modal_assets() {
+	if ( should_enable_subscriber_modal() && ! is_admin() ) {
+		wp_enqueue_style( 'subscribe-modal-css', plugins_url( 'subscribe-modal.css', __FILE__ ), array(), JETPACK__VERSION );
```

Comment:
You could add this new stylesheet to `concatenated_style_handles` in `class.jetpack.php` so the contents of the file can be added to the concatenated `jetpack.css` that's loaded for all connected Jetpack sites.

Reply:
I've added this. To confirm, I still enqueue the new stylesheet like normal too? Based on what I'm seeing, that seems to be case.

Reply:
Yes, you continue to enqueue the stylesheet as usual. We take care of dequeuing it when the concatenated version has to be enqueued instead.